### PR TITLE
feat:Added Collector for Kubernetes Services in Azure

### PR DESCRIPTION
**Pull request by kushZemoso**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List
any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The commit message/PR follows our guidelines

# **Other information**:

### CODE REVIEW

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Kubernetes/KubernetesAddOnEnabledRule.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+        }
+        String esUrl = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        String url = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        if (!StringUtils.isNullOrEmpty(url)) {
+            esUrl = url + "/azure_kubernetes/_search";
+        }
+        boolean isValid = false;
+        Map<String, Object> mustFilter = new HashMap<>();
+        mustFilter.put(PacmanRuleConstants.LATEST, true);
```
Comment (dheerajkholia-paladin) [inline jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Kubernetes/KubernetesAddOnEnabledRule.java:45]: To retrieve only the resource data shouldn't we add resourceid in the filer,as done for other policies.

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Kubernetes/KubernetesPrivateClusterEnableRule.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+        }
+        String esUrl = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        String url = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        if (!StringUtils.isNullOrEmpty(url)) {
+            esUrl = url + "/azure_kubernetes/_search";
+        }
+        boolean isValid = false;
+        Map<String, Object> mustFilter = new HashMap<>();
+        mustFilter.put(PacmanRuleConstants.LATEST, true);
```
Comment (dheerajkholia-paladin) [inline jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Kubernetes/KubernetesPrivateClusterEnableRule.java:44]: Should this contain filter for resourceId?

Comment (kushZemoso) [reply jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Kubernetes/KubernetesPrivateClusterEnableRule.java]: Done

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Kubernetes/KubernetesAddOnEnabledRule.java (hunk 1 above)
Comment (kushZemoso) [reply jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Kubernetes/KubernetesAddOnEnabledRule.java]: Done

**Review by dheerajkholia-paladin — APPROVED** [review]

**Review by santhosh-challa — APPROVED** [review]

**Pull request merged**